### PR TITLE
*: remove async batch snap

### PR DESCRIPTION
**Pull request by overvenus**
## What have you changed? (mandatory)

The async batch snapshot API is obsolete. The coprocessor and scheduler no longer use it, so it can be removed.

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Unit tests.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

## Refer to a related PR or issue link (optional)

#2107 and #3551

### CODE REVIEW

**Comment by BusyJay** [comment]
Why mark as DNM?

**Comment by overvenus** [comment]
It is based on https://github.com/tikv/tikv/pull/3551 which is not meged yet.

**Comment by overvenus** [comment]
PTAL, thanks!

File: benches/raftkv/mod.rs [hunk 1]
```diff
@@ -109,49 +100,35 @@ fn new_engine() -> (TempDir, Arc<DB>) {
     (dir, Arc::new(db))
 }
 
+// speed of light for async_snapshot
```
Comment (BusyJay) [inline benches/raftkv/mod.rs]: Strange comment.

**Pull request merged**